lib/sagas/lifecycle.js

```diff
@@ -181,6 +184,7 @@ function* lifecycle (action) {
       console.log('err2: ', err.toString())
 
       yield* fail(uid)
+      yield put(setWorkfowStatusPermanentFailure())
```

Comment:
sagas will receive this and do `store.dispatch` on its own. jumping into redux+sagas can be confusing for sure.